retry/lib/retry.dart

```diff
@@ -132,6 +133,10 @@ class RetryOptions {
       } on Exception catch (e) {
         if (attempt >= maxAttempts ||
             (retryIf != null && !(await retryIf(e)))) {
+          if (orElse != null) {
+            return await orElse(e);
+          }
```

Comment:
There are two kinds of exceptions here:
 * (A) Exceptions that trigger a retry -- those where `retryIf(e)` returns true.
 * (B) Exceptions that don't trigger a retry.

There are two reasons a call to `retry(fn, ...)` may throw:
 * (i) `fn` throws an exception of type (B),
 * (ii) `fn` throws an exception of type (A) enough times that the number of retries have been exhausted.

Do we think it's right to call `orElse` in both cases (i) and (ii)?

Reply:
Oh, this is an old one that is still open :)

I think we should only have an alternative callback for (ii), and with that in mind, `orElse` is not the best name for it. How about `afterExhausted` or `onAttemptsExhausted`?

Reply:
I think `orElse` is an okay name, but we might have to document the behavior -- if we want this at all.